enhancements/installer/bootstrapping-clusters-with-capi-providers.md

```diff
@@ -0,0 +1,646 @@
+
+For the initial implementation, CAPI providers will be vendored and built within the Installer repo:
+
+```bash
+$ tree cluster-api/providers -L 2
+cluster-api/providers
+├── aws
+│   ├── go.mod
+│   ├── go.sum
+│   ├── tools.go
+│   └── vendor
+├── azure
+│   ├── go.mod
+│   ├── go.sum
+│   ├── tools.go
+│   └── vendor
+...
+```
+
+This follows a similar pattern to the current Terraform provider implementation and build. This
+pattern will be used initially due to its simplicity and existing support. This pattern has
+drawbacks because changes to providers need to be merged upstream first and then vendored
+to the Installer. This aspect will be particularly problematic once CAPI providers are GA
+and we need to keep the Installer in-sync with other providers. Builds are also inefficient
+in that providers are always rebuilt, even when unchanged.
+
+##### Copying Dependencies from Container Images
+
+[TODO]
+
+#### Disk space and memory
+
+> NOTE: All tests are performed on a MacOS system at the time of writing.
+
+With the introduction of the Local Control Plane, new required components are being run during cluster creation.
+
+- `api-server` uses ~280MB and `etcd` ~52MB
+- Cluster API
+  - Core controller uses ~28MB
+  - Infrastructure controller (AWS, in testing) uses ~35MB
+
+When compared to Terraform, the runtime memory footprint is well-within the current usare or limitations (TODO: provide Terraform benchmark); the disk footprint increases by what's required of `etcd` which is around 128MB while creating an OpenShift cluster on AWS.
+
+#### Binary size
+
+> NOTE: All tests are performed on a MacOS system at the time of writing.
+
+The `openshift-install` resulting binary size, with Cluster API bundled binaries reaches ~515MB, of which, the bundled binaries account 228MB. By comparison, the binary available for download on `console.redhat.com` for MacOS, once upacked, is 732MB.
+
+The binaries are stored in `zip` format within the installer, and unpacked at runtime; from the `zip` output, the space savings are quite substantial:
+
+```
+updating: cluster-api (deflated 68%)
+updating: cluster-api-provider-aws (deflated 73%)
+updating: cluster-api-provider-azure (deflated 70%)
+updating: cluster-api-provider-azureaso (deflated 74%)
+updating: cluster-api-provider-gcp (deflated 73%)
+updating: cluster-api-provider-ibmcloud (deflated 72%)
+updating: cluster-api-provider-nutanix (deflated 69%)
+updating: cluster-api-provider-vsphere (deflated 72%)
+updating: etcd (deflated 64%)
+updating: kube-apiserver (deflated 72%)
+```
+
+## Design Details
+
+### Open Questions [optional]
+
+- UX design during install process as well as during failure (log collection). The Installer will dump
+(potentially prettified) controller logs. We expect that this question will become easier to answer further
+into the development process.
+
+- Whether to use downstream OpenShift-specific `kube-apiserver` and `etcd` dependencies and how to source them?
```

Comment:
We are required to use downstream builds, but not necessarily from our forks. However, if we don't use our forks, we have no ability to patch should we need to.

Reply:
Removed the open question as this has been resolved.